lib/pact/command/cont_command.ex

```diff
@@ -227,6 +231,36 @@ defmodule Kadena.Pact.ContCommand do
 
   def build(_module), do: {:error, [exec_command_request: :invalid_payload]}
 
+  @spec set_signers_from_keypair(t()) :: {:ok, t()}
+  defp set_signers_from_keypair(
+         %__MODULE__{
+           signers: signers
+         } = cmd_request
+       )
+       when signers != [],
+       do: {:ok, cmd_request}
+
+  defp set_signers_from_keypair(
+         %__MODULE__{
+           signers: signers,
+           keypairs: keypairs
+         } = cmd_request
+       ) do
+    case signers == [] && keypairs != [] do
+      true -> {:ok, set_signers(cmd_request)}
+      false -> {:ok, cmd_request}
+    end
+  end
+
+  @spec set_signers(t()) :: t()
+  defp set_signers(%__MODULE__{keypairs: key_pairs} = cmd_request) do
+    Enum.map(key_pairs, fn %KeyPair{pub_key: pub_key, clist: clist} ->
+      Signer.new(pub_key: pub_key, clist: clist)
+    end)
+    |> SignersList.new()
+    |> (&add_signers(cmd_request, &1)).()
+  end
+
```

Comment:
@Odraxs after answering the previous question. It is necessary to update/refactor these functions.